src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificatePdfGenerationService.java

```diff
@@ -0,0 +1,150 @@
+package ch.admin.bag.covidcertificate.service;
+
+import ch.admin.bag.covidcertificate.api.exception.CreateCertificateException;
+import ch.admin.bag.covidcertificate.api.request.pdfgeneration.AntibodyCertificatePdfGenerateRequestDto;
+import ch.admin.bag.covidcertificate.api.request.pdfgeneration.ExceptionalCertificatePdfGenerateRequestDto;
+import ch.admin.bag.covidcertificate.api.request.pdfgeneration.RecoveryCertificatePdfGenerateRequestDto;
+import ch.admin.bag.covidcertificate.api.request.pdfgeneration.RecoveryRatCertificatePdfGenerateRequestDto;
+import ch.admin.bag.covidcertificate.api.request.pdfgeneration.TestCertificatePdfGenerateRequestDto;
+import ch.admin.bag.covidcertificate.api.request.pdfgeneration.VaccinationCertificatePdfGenerateRequestDto;
+import ch.admin.bag.covidcertificate.api.request.pdfgeneration.VaccinationTouristCertificatePdfGenerateRequestDto;
+import ch.admin.bag.covidcertificate.api.response.CovidCertificateCreateResponseDto;
+import ch.admin.bag.covidcertificate.api.response.CovidCertificateResponseEnvelope;
+import ch.admin.bag.covidcertificate.service.document.PdfCertificateGenerationService;
+import ch.admin.bag.covidcertificate.service.domain.AbstractCertificatePdf;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import se.digg.dgc.encoding.BarcodeException;
+import se.digg.dgc.encoding.impl.DefaultBarcodeCreator;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import static ch.admin.bag.covidcertificate.api.Constants.CREATE_BARCODE_FAILED;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class CovidCertificatePdfGenerationService {
```

Comment:
To me a class name containing fromExisting would be more accurate. By reading the class name I'd know what endpoints are handled.